src/org/thoughtcrime/securesms/mms/AttachmentManager.java

```diff
@@ -383,17 +385,12 @@ public boolean isAttachmentPresent() {
   }
 
   public static void selectDocument(Activity activity, int requestCode) {
-    Permissions.with(activity)
-               .request(Manifest.permission.WRITE_EXTERNAL_STORAGE)
-               .ifNecessary()
-               .withPermanentDenialDialog(activity.getString(R.string.perm_explain_access_to_storage_denied))
-               .onAllGranted(() -> selectMediaType(activity, "*/*", null, requestCode))
-               .execute();
```

Comment:
I'm not removing the "recent photos" feature here, don't worry, I love this feature too  :heart_eyes:

This is about removing a permission request that is not necessary at this point.

This is hard to explain, so I suggest you try it out yourself (build the PR, revoke storage permission and see when you are asked for this permission). With this change, you won't be asked for storage permission when attaching a "File", but you are still asked when attaching from "Gallery". (we can also do an a/v at some point where I show you)

Reply:
on #2103 I use this `selectMediaType` thing, I used the permissions boilerplate just because it was used in other parts of the code, and I was afraid it was needed in older Android versions, but indeed at first I removed the storage permission and tried without this boilerplate and it also works so at least in Android 11 it is not needed, btw, would be good to merge #2103 first and then apply all this API updates because I am using the old ways in that PR

Reply:
We can choose the merging order as we like, it would be possible to use the new API in #2103 already without targeting API30.

But actually at first glance #2103 looks fine regarding the new API.

Reply:
the issue is that I know nothing about the new API



> But actually at first glance #2103 looks fine regarding the new API

at first glance I use a block of code exactly like this one you just removed, check https://github.com/deltachat/deltachat-android/pull/2103/files#diff-8079ced642052b025ab03eca06ffba4fc450a604e7eef4ac54db986000ff2637R150

Reply:
#2103 can be merged later but will need your careful review so I don't mess something and reintroduce things like that

Reply:
Sure, I'll do that :+1::yellow_heart: It's not that I know much about the new API, either, though, I mainly read some things about it online, tested where things break when targeting API 30 and then tried to fix them.

So, the main thing will be testing #2103 and if things break do trial and error until it's fixed :joy: okay, and read the documentation.